fsspec/implementations/memory.py

```diff
@@ -53,7 +53,7 @@ def ls(self, path, detail=False, **kwargs):
                     out.append(
                         {
                             "name": p2,
-                            "size": self.store[p2].getbuffer().nbytes,
```

Comment:
Since `.getvalue()` returns a view, how does `len(self.store[p2].getvalue())` compare? It's certainly simpler looking.

Reply:
Ok, so here were my considerations there:

I also like the look of that better, *and* it's even more performant if no buffer is extant.  However, if the user has a buffer somewhere, and the file's large, it can be sluggish.  

No writing has occurred and no buffer extant:
  * `_flen`: 228 ns
  * `getbuffer`: 169 ns
  * `getvalue`: 82.9 ns

Writing has occurred, or there is a buffer extant:
  * `_flen1: 228 ns
  * `_getbuffer`: 168 ns
  * `_getvalue`: 2.86 **ms**  < -- Ouch
  
  So the conclusion is:
  * `len(f.getbuffer())` is consistently a little on the faster side
  * `len(f.getvalue())` is faster when no writes have occurred, but terrible if dealing with a large file that has been written to or has a buffer extant
  * `_flen(f)` is consistently a little slower, but is consistent, and still decently fast.
  
That's why I went with `_flen` -- but yeah, it's not as appealing stylistically.  That's also why I considered making a magic `MemoryFile.__len__` so we could do `len(f)` (or more specifically, `len(self.store[p2])` -- but I didn't want to push for that since filelike objects don't typically support `len()`, and I thought that might be considered too weird of an API change.

Reply:
@martindurant 

That's on a 30 MB file, btw.

While I'm perfectly happy using `getvalue()` here, I suspect that people who have been doing writes will be disappointed with the cost of getting length (2860000ns vs 230ns is a big leap, even if occasional), plus there's the memory cost of a copy, albeit a temporary one.  I think the ~230ns range is pretty much fine, as it allows ~6k size checks per ms, and avoids copy.

Let me know your thoughts.  I really think the seek/tell method is best, but I'm happy to frame it differently.

A simple thing that would fit in with what's already there would be to implement `MemoryFile.size` as a property instead of as a variable.  This has the added benefit of providing the right size if someone has acquired the file and modified it, and then does a `memfs.ls('/containing_dir/', detail=True)`.

Let me know what you think..

Reply:
Let's replace both other uses of `_flen`, in that case. I might still add that function to fsspec.utils at another time, though, since it's an obvious thing one might want to do in some situations.

Reply:
With a `MemoryFile.size` property that does the same thing, yes?

Reply:
Happy to have a property, yes, but let's do it without `seek`